tests/numpy/test_sampled_from.py

```diff
@@ -0,0 +1,42 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+
+from hypothesis import given
+from tests.common.utils import checks_deprecated_behaviour
+from hypothesis.strategies import sampled_from
+
+a_numpy_array = np.array([1, 2, 3])
+
+a_multi_dimensional_numpy_array = np.array([[1, 2, 3], [4, 5, 6]])
```

Comment:
Shouldn't we use hypothesis to generate numpy arrays and sample from those to test this feature?

Reply:
For instance, your test will fail for this case:
```
>>> a = np.array([[1,2,3], [], [4,5,6]])
>>> a
array([[1, 2, 3], [], [4, 5, 6]], dtype=object)
>>> t = tuple(a)  # what the patch does with ndarrays
>>> t[0] in a
False
```

Reply:
> For instance, your test will fail for [object arrays]

You're right, but I think that's "works as intended":

    >>> a = np.array([[]])
    >>> a[0] in a
    False

Numpy arrays with object dtype have some odd behaviour around membership testing, but that's not our fault.  `tuple(a)[0] == a[0]` (assuming it's non-NaN), and that's good enough for me!

Reply:
Yes. My point was that current test method doesn't suites well for possible cases.

Reply:
Can you suggest an alternative test then?